clients/drcachesim/tools/schedule_stats.h

```diff
@@ -73,9 +75,51 @@ class schedule_stats_t : public analysis_tool_t {
     std::string
     parallel_shard_error(void *shard_data) override;
 
+    // Simple binning histogram for instrs-per-switch distribution.
+    class histogram_t {
```

Comment:
> This feature is structured to allow replacing this simple histogram with a more sophisticated version in subclasses.

From other proposed internal changes, I see that the plan is to create a derived class that overrides the various histogram_t APIs. But that may silently break things if histogram_t implements some new public API which is not overridden by the derived class.

It would be better to have a virtual base-class that can be implemented by both histogram_t and other more-sophisticated implementations. This way, if there's some new public API, that will show up as a clear build failure up-front.

Reply:
> virtual base-class

I mean an abstract base-class with pure virtual functions.

Reply:
Done.